aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/NotificationChannelHelper.java

```diff
@@ -0,0 +1,58 @@
+package software.amazon.codeguruprofiler.profilinggroup;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableSet;
+import software.amazon.awssdk.services.codeguruprofiler.CodeGuruProfilerClient;
+import software.amazon.awssdk.services.codeguruprofiler.model.AddNotificationChannelsRequest;
+import software.amazon.awssdk.services.codeguruprofiler.model.Channel;
+import software.amazon.awssdk.services.codeguruprofiler.model.EventPublisher;
+import software.amazon.awssdk.services.codeguruprofiler.model.RemoveNotificationChannelRequest;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+
+public class NotificationChannelHelper {
+    private NotificationChannelHelper() {
+        // prevent instantiation
+    }
+
+    public static void addChannelNotifications(String pgName, List<software.amazon.codeguruprofiler.profilinggroup.Channel> channels,
+                                               AmazonWebServicesClientProxy proxy, CodeGuruProfilerClient profilerClient) {
+        AddNotificationChannelsRequest.Builder addNotificationChannelsRequest = AddNotificationChannelsRequest.builder()
+                .profilingGroupName(pgName);
+        addNotificationChannelsRequest.channels(channels.stream().map(channel -> Channel.builder().uri(channel.getChannelUri())
+                .eventPublishers(ImmutableSet.of(EventPublisher.ANOMALY_DETECTION))
+                .id(channel.getChannelId()).build()).collect(Collectors.toList()));
+
+        proxy.injectCredentialsAndInvokeV2(addNotificationChannelsRequest.build(), profilerClient::addNotificationChannels);
+    }
+
+    public static void addChannelNotification(String pgName, Channel channel, AmazonWebServicesClientProxy proxy, CodeGuruProfilerClient profilerClient) {
+        addChannelNotifications(pgName, Collections.singletonList(software.amazon.codeguruprofiler.profilinggroup.Channel.builder()
+                .channelId(channel.id())
+                .channelUri(channel.uri())
+                .build()), proxy, profilerClient);
+    }
```

Comment:
Minor: The interaction between `addChannelNotification` and `addChannelNotifications` is slightly weird, because we convert from an SDK channel into a "this package channel" back into an SDK channel.

I suggest adding an extra overload to `addChannelNotifications` that can receive the SDK channels, and that would avoid this conversion. (The `addChannelNotifications` with the "this package channel" instance would just convert the instances and then call the other one)